### DISCUSSION TITLE
Have an issue on Ubuntu Linux

### DISCUSSION THREAD

**Discussion by furious1232** [ORelio/Minecraft-Console-Client#2374] [Q&A]
I been using cc for a few years now on windows and now i am on a server that has alt limits so i now have 2 Ubuntu Linux VPS so i can bypass the limit.
So i have installed wine fully and checked it works and it does. I followed this video https://www.youtube.com/watch?v=Dolxkfk-P-c

The MCC works for a few secs just after its connects to the server the chat lags out and it then takes a min and relogs and does that loop. 
The MCC i use is the one i had for windows and it works perfectly on windows (Been using the same one once since the migration) 
Is this the reason?

**Comment by ReinforceZwei** [comment]
MCC now has native Linux support. You don't need wine to run MCC. Download Linux version binary from [release](https://github.com/MCCTeam/Minecraft-Console-Client/releases).

**Reply by furious1232** [reply]
So I have installed the Linux version along with .NET SDK. And it still doesn't work.
 What do I even open the MCC with for it to launch?

**Reply by BruceChenQAQ** [reply]
What output do you get when you run `./MinecraftClient` directly?

**Reply by furious1232** [reply]
![image](https://user-images.githubusercontent.com/120287293/206899000-13344fcc-fa0f-40dc-a245-44fca5815220.png)

**Reply by BruceChenQAQ** [reply]
> ![image](https://user-images.githubusercontent.com/120287293/206899000-13344fcc-fa0f-40dc-a245-44fca5815220.png)

I mean, just run the `MinecraftClient` binary file that you downloaded and extracted.

**Reply by furious1232** [reply]
Just opens a notepad.

**Reply by BruceChenQAQ** [reply]
> Just opens a notepad.

What command did you use to run it?

Try using the `file` command to see what type of file it is: `file /path/to/MinecraftClient`
and see what the output will be.

**Reply by furious1232** [reply]
I just clicked on the MCC I downloaded. 
I just opened the terminal and ran that.
![image](https://user-images.githubusercontent.com/120287293/206899546-92c32cb5-bded-420e-b8ec-65c14a9b4981.png)

**Reply by milutinke** [reply]
> I just clicked on the MCC I downloaded. I just opened the terminal and ran that. ![image](https://user-images.githubusercontent.com/120287293/206899546-92c32cb5-bded-420e-b8ec-65c14a9b4981.png)

Have you extracted it from the zip file?
From the command output, the file doesn't exists
The file should have execution permissions, but if it is there and won't execute, try with `chmod +x MinecraftClient` just in case, and then try to run it.

**Reply by BruceChenQAQ** [reply]
> I just clicked on the MCC I downloaded. I just opened the terminal and ran that. ![image](https://user-images.githubusercontent.com/120287293/206899546-92c32cb5-bded-420e-b8ec-65c14a9b4981.png)

If the extracted file is stored at `/root/downlads/cc/MinecraftClient`
try `/root/downlads/cc/MinecraftClient` and `file /root/downlads/cc/MinecraftClient` in terminal

**Reply by furious1232** [reply]
Yes, I extracted the file.

I just ran the command and it generated the config but I don't see it in the file. 
![image](https://user-images.githubusercontent.com/120287293/206899933-76b4ee11-3a38-473e-a76a-224d7816076e.png)

**Reply by furious1232** [reply]
Does anyone want to voice call on discord to help me? May be easier and quicker.

**Reply by BruceChenQAQ** [reply]
> Yes, I extracted the file.
> 
> I just ran the command and it generated the config but I don't see it in the file. ![image](https://user-images.githubusercontent.com/120287293/206899933-76b4ee11-3a38-473e-a76a-224d7816076e.png)

The configuration file should be in the same directory as the `MinecraftClient` binary.

**Reply by furious1232** [reply]
I only have this the one that you would launch it with if you could click on it.
![image](https://user-images.githubusercontent.com/120287293/206900118-c1b7802d-1b74-4ccd-ab64-c5013f44c85b.png)

**Reply by BruceChenQAQ** [reply]
> I only have this the one that you would launch it with if you could click on it.

Check the directory where the terminal was located when you executed the command `/root/downlads/cc/MinecraftClient`.
It may be in the folder `/root`.

**Comment by furious1232** [comment]
I just used the terminal that you can search for.

**Comment by BruceChenQAQ** [comment]
Try running these:
```sh
cd /root/downlads/cc/
./MinecraftClient
```

The configuration file should then be generated as `/root/downlads/cc/MinecraftClient.ini`.

**Reply by furious1232** [reply]
Ok, it was in the root folder. So I moved it to where I wanted it. How do I start the MCC from that folder then?
![image](https://user-images.githubusercontent.com/120287293/206900725-860d1b61-a781-4b23-9a46-ba2579edce71.png)

**Reply by BruceChenQAQ** [reply]
```sh
cd /root/downlads/cc/
./MinecraftClient
```
OR
```sh
/root/downlads/cc/MinecraftClient /root/downlads/cc/MinecraftClient.ini
```

**Reply by furious1232** [reply]
Is there a way to like click on the MinecraftClient to run it or do I have to run a command? If there is a way to click on it how do i do that? Its just hard for me to use Linux

**Reply by BruceChenQAQ** [reply]
I am not familiar with the GUI of Linux, you can ask others how to run it from the GUI.

**Reply by furious1232** [reply]
Ok, thanks for the help you can do. This works now just don't fully know how to use it on Linux. Well, time for me to just mess with it until I understand it. Thanks for the help you can do.

**Reply by breadbyte** [reply]
> Is there a way to like click on the MinecraftClient to run it or do I have to run a command?

I don't have a Linux GUI available to test this out, but from what I can recall, you can right click the MinecraftClient application, open Properties, and enable the execute permission. You should be able to open the file by double-clicking on it.
